### PR TITLE
[feat ]Add CoverageChecker to setInterruptHandler in util/Promise.scala

### LINKED ISSUE #16: Add CoverageChecker to setInterruptHandler in util/Promise.scala
The DIY code coverage tool needs to be implemented in setInterruptHandler@510-532@./com/twitter/util/Promise.scala. Identify all the different branches for the function, and make sure that they are covered by the CoverageChecker.

### PR DESCRIPTION
**Pull request by Zodbigt**
This PR adds coverage checking has been added to the method 'setInterruptHandler' in util/Promise.scala. Resolves #16 

[Issue: #16]

### CODE REVIEW

File: util-core/src/test/scala/com/twitter/util/DurationTest.scala [hunk 1]
```diff
@@ -296,7 +296,7 @@ class DurationTest extends { val ops: Duration.type = Duration } with TimeLikeSp
 
   "CoverageChecker" should {
     "print map" in {
-      println(CoverageChecker.map.getOrElse("parse", sys.error(s"unexpected key")).mkString("[",", ", "]"))
+      println("[Coverage - parse] - " + CoverageChecker.map.getOrElse("parse", sys.error(s"unexpected key")).mkString("[",", ", "]"))
```
Comment (stevensdavid) [inline util-core/src/test/scala/com/twitter/util/DurationTest.scala]: this shouldn't be here!

File: util-core/src/main/scala/com/twitter/util/Promise.scala [hunk 2]
```diff
@@ -509,25 +509,55 @@ class Promise[A] extends Future[A] with Promise.Responder[A] with Updatable[Try[
...
+        // ignore
+        CoverageChecker.reached(funcName, 7)
+      }
 
-    case p: Promise[A] /* Linked */ => p.setInterruptHandler(f)
+      case p: Promise[A] /* Linked */ => {
+        CoverageChecker.reached(funcName, 8)
+        p.setInterruptHandler(f)
+      }
+    }
```
Comment (stevensdavid) [inline util-core/src/main/scala/com/twitter/util/Promise.scala:561]: I guess that technically there's a default case `case _: => ()` that should be tracked?

**Review by stevensdavid — CHANGES_REQUESTED** [review]

File: util-core/src/main/scala/com/twitter/util/Promise.scala (hunk 2 above)
Comment (felixlut) [reply util-core/src/main/scala/com/twitter/util/Promise.scala:561]: No there shouldn't be one. See issue #27

File: util-core/src/test/scala/com/twitter/util/DurationTest.scala (hunk 1 above)
Comment (Zodbigt) [reply util-core/src/test/scala/com/twitter/util/DurationTest.scala]: Ill fix that!

**Review by stevensdavid — APPROVED** [review]

**Pull request merged**